### PR TITLE
Additional Gif Builder API

### PR DESCRIPTION
**Pull request by novedevo**
not sure whether this is something you want to merge upstream. there's definitely things that should be changed before merging, note the comments I left in the pr review thingy

### CODE REVIEW

File: gifed/src/block/palette.rs [hunk 1]
```diff
@@ -48,13 +48,12 @@ impl Palette {
...
 
-	pub fn from_color<C: AsRef<Color>>(&self, color: C) -> Option<u8> {
-		for (i, &c) in self.table.iter().enumerate() {
-			if c == *color.as_ref() {
-				return Some(i as u8);
-			}
-		}
-		None
+	pub fn from_color(&self, color: Color) -> Option<u8> {
+		self.table
```
Comment (novedevo) [inline gifed/src/block/palette.rs:52]: i should have made this self.table.iter().position(|c| **c == color)

File: gifed/src/gif_builder.rs [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+		Self {
+			image_indices,
+			interval: None,
+			palette,
+		}
+	}
+}
+
+impl Frame {
+	///
```
Comment (novedevo) [inline gifed/src/gif_builder.rs]: why did I leave this here

File: gifed/src/gif_builder.rs [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+		let images = frames.into_iter().map(|frame| {
+			let Frame {
+				interval,
+				image_indices,
+				palette,
+			} = frame;
+
+			let delay = interval
+				.map(|interval| interval * 10)
```
Comment (novedevo) [inline gifed/src/gif_builder.rs]: this is a bug that should be fixed

File: gifed/src/lib.rs [hunk 4]
```diff
@@ -1,18 +1,25 @@
...
 pub use writer::EncodeError;
 
-pub use color::Color;
 pub use gif::{Gif, Image};
 pub use lzw::LZW;
 
+#[cfg(feature = "rgb")]
+pub type Color = rgb::RGB8;
+#[cfg(not(feature = "rgb"))]
+mod color;
```
Comment (novedevo) [inline gifed/src/lib.rs:19]: should this be pub mod color

File: gifed/Cargo.toml [hunk 5]
```diff
@@ -1,18 +1,24 @@
...
-edition = "2018"
+authors = ["gennyble <gen@nyble.dev>", "novedevo <devon@nove.dev>"]
+edition = "2021"
 license = "ISC"
 description = "Gif encoding and decoding with fine control"
 repository = "https://github.com/genuinebyte/gifed"
 
 [dependencies]
 bitvec = "1.0.1"
+colorsquash = { git = "https://github.com/novedevo/colorsquash", version = "0.2.0", optional = true }
```
Comment (novedevo) [inline gifed/Cargo.toml]: this is a lil questionable. this PR depends on a version of colorsquash with RGB support, which is currently only my fork. if you merge the other PR on colorsquash then you can change this to your own colorsquash or something?

Comment (gennyble) [reply gifed/Cargo.toml]: colorsquash doesn't seem to do well here, so I've switched the quantization to `color_quant` and dropped colorsquash.

One day I'd like to allow selecting quantizer with a trait and dependency injection *(don't like this name, I don't think it's very clear; "here gifed, it's a quantizer use this one instead")* and have feature flags, but that day is not today.

**Comment by gennyble** [comment]
i think it's something good to have upstream. I like `VideoGif` as a name more than `GifBuilder` because it limits you to a video-like gif *(more so than it used to, now, too)*. Thank you friend <3

**Pull request merged**